Library/Formula/elftoolchain.rb

```diff
@@ -0,0 +1,69 @@
+class Elftoolchain < Formula
+  desc "Compilation tools and libaries for handling the ELF file format"
+
+  homepage "http://elftoolchain.sourceforge.net/"
+  url "https://downloads.sourceforge.net/project/elftoolchain/Sources/elftoolchain-0.6.1/elftoolchain-0.6.1.tgz"
+  sha256 "f4dc6e2a820f146658d6b5f9a062a2e676d0715aca654f55866c60e0025561eb"
+
+  depends_on :bsdmake => :build
+  depends_on "libarchive"
+
+  keg_only <<-EOS.undent
+    The tools provided conflict with tool names from the normal toolchain, such
+    as nm, and are not replacements for them.
+  EOS
+
+  # elftoolchain is in general pretty unaware of OS X / Darwin. The next 4
```

Comment:
We're going to need these patches reported upstream and those upstream bug reports linked in the formula. We rarely accept heavily patched new submissions as we end up carrying and supporting patches indefinitely otherwise.